onnxruntime/core/session/inference_session_utils.cc

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/session/inference_session_utils.h"
+
+namespace onnxruntime {
+
+//---------------------
+//--- local helpers ---
+//---------------------
+
+//--------------------------------------------
+//--- session options related helpers ---
+//--------------------------------------------
+// Below are some helpers that will be used to set corresponding session option values
+
+static Status
+SetIntraOpNumThreads(SessionOptions& session_options,
+                     int value,
+                     const logging::Logger& logger) {
+  if (value < 0) {
+    LOGS(logger, ERROR) << "Unsupported value for intra_op_num_threads: " << value;
+    return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Unsupported value for intra_op_num_threads: ", value);
+  }
+
+  LOGS(logger, INFO) << "Setting intra_op_num_threads to " << value;
+  session_options.intra_op_num_threads = value;
+  return Status::OK();
+}
+
+static Status SetInterOpNumThreads(SessionOptions& session_options,
+                                   int value,
+                                   const logging::Logger& logger) {
+  if (value < 0) {
+    LOGS(logger, ERROR) << "Unsupported value for inter_op_num_threads: " << value;
+    return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Unsupported value for inter_op_num_threads: ", value);
+  }
+
+  LOGS(logger, INFO) << "Setting inter_op_num_threads to " << value;
+  session_options.inter_op_num_threads = value;
+  return Status::OK();
+}
+
+static Status SetExecutionMode(SessionOptions& session_options,
```

Comment:
The meanings of 0's and 1's will need to be documented somewhere. Not just for this function, but for others as well.

Reply:
The meaning/interpretation of all the valid values for each session option is documented in our OneNote - a copy of which will be shared to the customer. Will that do ?